In [6]:
from math import pi
import sympy
from sympy import *

r, R_c = symbols('r R_c', real='True', positive='True')

Question 1:
$$\rho = 150 \frac{g}{cm^{3}}$$
$$T = 1.50 \times 10^{7} \, K$$
$$X \approx 0.7$$
The energy producing region of the Sun (the "core") extends from $r = 0$ to $r= R_{c}$. The energy from the core comes completely from the $p - p$ chain with an energy generation rate:
$$q_{pp} = \frac{2.4 \times 10^{4} \rho X^{2}}{T_{9}^{2/3}} e^{-3.380/{T_{9}^{1/3}}}$$
$$L_{nuc} = \int_{0}^{M} q\, dm$$
$$L_{nuc} = 4 \pi \rho \int_{0}^{R_c} q r^{2}\, dr \rightarrow \frac{9.6 \times 10^{4} \rho^{2} X^{2} \pi}{T_{9}^{2/3}} e^{-3.380/{T_{9}^{1/3}}} \int_{0}^{R_c} r^{2}\, dr$$

In [5]:
simplify((9.6 * 10**4 * 150**2 * 0.7**2 * pi)/((0.015)**Rational(2,3)) * exp((-3.380)/((0.015)**Rational(1,3))))

19430.4958007566*pi

In [7]:
integrate(r**2,(r,0, R_c))

R_c**3/3

$$L_{nuc} = 19430.5 \pi [\frac{R_{c}^{3}}{3}]$$
We know that the Sun's luminosity is, $L = 3.826 \times 10^{33}$ ergs/second:
$$3.826 \times 10^{33} = \frac{19430.5 R_{c}^{3} \pi}{3}$$

In [13]:
x = simplify((3.826 * 10**33 * 3)/ (19430.5 * 3.14159265358))
simplify(x**(Rational(1,3)))

5728981992.02917

Therefore,
$$R_{c} = 5.72898 \times 10^{9}\, cm$$

Question 2 Part(a):
$$^{16}_{8}O + ^{16}_{8}O \rightarrow ^{31}_{16}S + n$$
Conservation of baryon number

Question 2 Part(b):
$$^{16}_{8}O + ^{16}_{8}O \rightarrow ^{32}_{16}S + \gamma$$
Conservation of momentum

Question 2 Part(c):
$$^7_4Be + e^- \rightarrow ^7_3Li + \nu$$
Conservation of lepton number

Question 2 Part(d):
$$^{15}_{8}O \rightarrow ^{15}_{7}N + p + \nu$$
Conservation of baryon number

Question 4 Part(a):
$$\lambda = \frac{h}{p};\, p = \mu_{p} * \sqrt{\frac{3 k_{b} T}{\mu_{p}}}$$

In [23]:
simplify((6.626 * 10**(-34)) / (8.35 * 10**(-28) * ((3 * 1.38*10**(-23) * 1.5 * 10**7)/(8.35 * 10**(-28)))**Rational(1,2)))

9.20157973340926e-13

Therefore,
$$\lambda = 9.202 \times 10^{-13}\, m$$

Question 4 Part(b):
$$P \approx e^{\frac{-r_{0}}{\lambda}} \rightarrow P \approx e^{-constant / v}$$
$$T_{1} = 10^{7}\, K;\, T_{2} = 1.5 \times 10^{7}\, K$$

In [27]:
velocity_1 = simplify(((3 * 1.38*10**(-23) * 1 * 10**7)/(8.35 * 10**(-28)))**Rational(1,2))
velocity_2 = simplify(((3 * 1.38*10**(-23) * 1.5 * 10**7)/(8.35 * 10**(-28)))**Rational(1,2))
simplify((exp(-1 / velocity_1))/(exp(-1/ velocity_2)))

0.999999739392343

Thus,
$$\frac{P_1}{P_2} = \frac{e^{-1/v_1}}{e^{-1/v_2}} = 0.9999$$